pkg/js/validator/validate-store.ts

```diff
@@ -0,0 +1,976 @@
+  }
+  return true;
+}
+
+function getRelationReferenceString(relationReference: RelationReference) {
+  let relationReferenceString = relationReference.type;
+  if (relationReference.wildcard) {
+    relationReferenceString += ":*";
+  } else if (relationReference.relation) {
+    relationReferenceString += `#${relationReference.relation}`;
+  }
+
+  if (relationReference.condition) {
+    relationReferenceString += `' with '${relationReference.condition}`;
+  }
+
+  return relationReferenceString;
+}
+
+function validateTypeRestrictions(
+  tuple: TupleKey,
+  typeDefs: TypeDefinition[],
+  conditions: Record<string, Condition> | undefined,
+  instancePath: string,
+): boolean {
+  validateTuple.errors = validateTuple.errors || [];
+
+  const mappedTuple = mapTuple(tuple);
+  const object = tuple.object.split(":")[0];
+  const type = typeDefs.filter((t) => t.type === object)[0];
+
+  const userTypes = type?.metadata?.relations?.[tuple.relation].directly_related_user_types;
+
+  if (
+    !userTypes?.filter(
+      (userType) =>
+        userType.type === mappedTuple.type && // type matches
+        !!mappedTuple.wildcard === !!userType.wildcard && // and the wildcard matches (either both true or both false)
+        userType.relation === mappedTuple.relation && // and the relation matches
+        userType.condition === mappedTuple.condition, // and the condition matches
+    ).length
+  ) {
+    validateTuple.errors.push(userNotTypeRestriction(getRelationReferenceString(mappedTuple), tuple, instancePath));
+    return false;
+  }
+
+  if (mappedTuple.condition && conditions) {
+    // Check parameters for matching condition
+    return validateConditionParams(tuple, mappedTuple.condition, conditions, instancePath);
+  }
+  return true;
+}
+
+function validateConditionExists(
+  tuple: TupleKey,
+  conditions: Record<string, Condition> | undefined,
+  instancePath: string,
+): boolean {
+  if (!tuple.condition) {
+    return true;
+  }
+
+  validateTuple.errors = validateTuple.errors || [];
+
+  // Condition on tuple not found
+  if (!conditions || !conditions[tuple.condition.name]) {
+    validateTuple.errors.push(conditionDoesntExist(tuple, instancePath));
+    return false;
+  }
+  return true;
+}
+
+function validateConditionParams(
+  tuple: TupleKey,
+  condition: string,
+  conditions: Record<string, Condition>,
+  instancePath: string,
+): boolean {
+  validateTuple.errors = validateTuple.errors || [];
+
+  if (tuple.condition && tuple.condition.context && conditions[condition].parameters) {
+    for (const param of Object.keys(tuple.condition.context)) {
+      if (!conditions[condition].parameters![param]) {
+        validateTuple.errors.push(notAParameter(param, tuple, instancePath));
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+// Validation for tuples
+const validateTuple: SchemaValidateFunction = function (
+  this: { jsonModel: AuthorizationModel },
+  tuple: TupleKey,
+  cxt: { instancePath: string },
```

Comment:
```suggestion
  ctx: { instancePath: string },
```

Reply:
I've changed over the names - `cxt` is the parameter name from ajv. 

https://ajv.js.org/keywords.html#schema-compilation-context